includes/Core/Util/Entity_Factory.php

```diff
@@ -0,0 +1,458 @@
+	 */
+	private static function create_entity_for_posts_blog() {
+		$post_id = (int) get_option( 'page_for_posts' );
+		if ( ! $post_id ) {
+			return null;
+		}
+
+		$post = get_post( $post_id );
+		if ( ! $post ) {
+			return null;
+		}
+
+		return new Entity(
+			get_permalink( $post ),
+			array(
+				'type'  => 'blog',
+				'title' => $post->post_title,
+				'id'    => $post->ID,
+			)
+		);
+	}
+
+	/**
+	 * Creates the entity for the front page blog archive.
+	 *
+	 * This method should only be used when the front page is set to display the
+	 * blog archive, i.e. is not technically a post itself.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Entity The entity for the front blog archive.
+	 */
+	private static function create_entity_for_front_blog() {
+		return new Entity(
+			user_trailingslashit( home_url() ),
+			array(
+				'type'  => 'blog',
+				'title' => __( 'Home' ),
+			)
+		);
+	}
+
+	/**
+	 * Creates the entity for a given term object, i.e. for a taxonomy term archive.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param WP_Term $term A WordPress term object.
+	 * @return Entity The entity for the term.
+	 */
+	private static function create_entity_for_term( WP_Term $term ) {
+		// See WordPress's `get_the_archive_title()` function for this behavior. The strings here intentionally omit
+		// the 'google-site-kit' text domain since they should use WordPress core translations.
+		switch ( $term->taxonomy ) {
+			case 'category':
+				$title  = _x( 'Category:', 'category archive title prefix' );
+				$title .= " {$term->name}";
+				break;
+			case 'post_tag':
+				$title  = _x( 'Tag:', 'tag archive title prefix' );
+				$title .= " {$term->name}";
+				break;
+			case 'post_format':
+				switch ( $term->slug ) {
+					case 'post-format-aside':
+						$title = _x( 'Asides', 'post format archive title' );
+						break;
+					case 'post-format-gallery':
+						$title = _x( 'Galleries', 'post format archive title' );
+						break;
+					case 'post-format-image':
+						$title = _x( 'Images', 'post format archive title' );
+						break;
+					case 'post-format-video':
+						$title = _x( 'Videos', 'post format archive title' );
+						break;
+					case 'post-format-quote':
+						$title = _x( 'Quotes', 'post format archive title' );
+						break;
+					case 'post-format-link':
+						$title = _x( 'Links', 'post format archive title' );
+						break;
+					case 'post-format-status':
+						$title = _x( 'Statuses', 'post format archive title' );
+						break;
+					case 'post-format-audio':
+						$title = _x( 'Audio', 'post format archive title' );
+						break;
+					case 'post-format-chat':
+						$title = _x( 'Chats', 'post format archive title' );
+						break;
+				}
+				break;
+			default:
+				$tax   = get_taxonomy( $term->taxonomy );
+				$title = sprintf(
+					/* translators: %s: Taxonomy singular name. */
+					_x( '%s:', 'taxonomy term archive title prefix' ),
+					$tax->labels->singular_name
```

Comment:
Yeah, it just struck me as off – a specific comment for this line is not necessary I don't think. Any time we reference a section of code copied or based on core, it would be good to be extra clear as to whether it is a copy or adaptation, and where it starts and ends. This reduces the need for cross-referencing. Not a big deal though 👍